Sources/MongoSwift/BSON/Document.swift

```diff
@@ -113,6 +113,20 @@ public struct Document: ExpressibleByDictionaryLiteral, ExpressibleByArrayLitera
      * - Returns: a new `Document`
      */
     public init(arrayLiteral elements: BsonValue?...) {
+        self.init(elements)
```

Comment:
Note: `convenience` was not used here as it's only for classes and Document is a struct.

Reply:
The bottom example of https://developer.apple.com/documentation/swift/expressiblebyarrayliteral shows an array literal initializer delegating to another `init` method:

```swift
extension OrderedSet: ExpressibleByArrayLiteral {
    init(arrayLiteral: Element...) {
        self.init()
        for element in arrayLiteral {
            self.append(element)
        }
    }
}
```

That documentation goes into some detail about why an Array isn't accepted in place of an array literal, but makes no claims about the opposite. I think we're in the clear with this.

The official docs for [`Set.init(arrayLiteral)`](https://developer.apple.com/documentation/swift/set/1539368-init) also show it annotated as a `convenience` initializer, from which we can reasonably conclude that it delegates to another `init` method.

Reply:
👍